src/main/java/com/ekart/thunderbolt/service/ProductService.java

```diff
@@ -0,0 +1,76 @@
+package com.ekart.thunderbolt.service;
+
+import com.ekart.thunderbolt.DAO.ProductDAO;
+import com.ekart.thunderbolt.entity.Product;
+import com.ekart.thunderbolt.mapper.ProductMapper;
+import com.ekart.thunderbolt.repository.ProductRepository;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
```

Comment:
Is this needed?

Reply:
@AllArgsConstructor is used to autowire dependencies.